JetMETCorrections/Type1MET/interface/JetCleanerForType1METT.h

```diff
@@ -0,0 +1,211 @@
+#ifndef JetMETCorrections_Type1MET_JetCleanerForType1METT_h
+#define JetMETCorrections_Type1MET_JetCleanerForType1METT_h
+
+/** \class JetCleanerForType1METT
+ *
+ * Clean jets for MET corrections and uncertainties (pt/eta/EM fraction and muons)
+ * apply also JECs
+ *
+ * NOTE: class is templated to that it works with reco::PFJets as well as with pat::Jets of PF-type as input
+ *
+ * \authors Matthieu Marionneau ETH
+ *
+ *
+ *
+ */
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/METReco/interface/CorrMETData.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+
+#include "JetMETCorrections/Type1MET/interface/JetCorrExtractorT.h"
+
+#include <string>
+#include <type_traits>
+
+
+namespace JetCleanerForType1MET_namespace
+{
+  template <typename T, typename Textractor>
+    class InputTypeCheckerT
+  {
+  public:
+
+    void operator()(const T&) const {} // no type-checking needed for reco::PFJet input
+    bool isPatJet(const T&) const {
+      return std::is_base_of<class pat::Jet, T>::value;
+    }
+  };
+
+  template <typename T>
+    class RawJetExtractorT // this template is neccessary to support pat::Jets
+    // (because pat::Jet->p4() returns the JES corrected, not the raw, jet momentum)
+    // But it does not handle the muon removal!!!!! MM
+    {
+    public:
+
+      RawJetExtractorT(){}
+      reco::Candidate::LorentzVector operator()(const T& jet) const
+	{
+	  return jet.p4();
+	}
+    };
+}
+
+
+template <typename T, typename Textractor>
+class JetCleanerForType1METT : public edm::stream::EDProducer<>
+{
+ public:
+
+  explicit JetCleanerForType1METT(const edm::ParameterSet& cfg)
+    : moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+    offsetCorrLabel_(""),
+    skipMuonSelection_(0)
+      {
+	token_ = consumes<std::vector<T> >(cfg.getParameter<edm::InputTag>("src"));
+
+	if ( cfg.exists("offsetCorrLabel") ) {
```

Comment:
cfg.exists can be used only in exceptional cases (e.g. there is a large number of existing configs to be modified or the parameter is interpreted by a remotely called helper class).
Please use fillDescriptions method 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp